lib/fizzy/parser_expr.cpp

```diff
@@ -76,6 +93,23 @@ void update_caller_frame(ControlFrame& frame, const FuncType& func_type)
     frame.stack_height += stack_height_change;
 }
 
+void validate_result_count(const ControlFrame& frame)
+{
+    if (!frame.unreachable)
```

Comment:
perhaps early return
```suggestion
    if (frame.unreachable)
        return;
```

Reply:
Done.